### PR TITLE
Remove references to the default credentials of admin:admin

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
Remove references to the default credentials of admin:admin and changes the CI according to default admin credentials changes for the 2.12.0 release.

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: guides/index_lifecycle.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ This guide covers OpenSearch Golang Client API actions for Index Lifecycle. You'
 
 ## Setup
 
-In this guide, we will need an OpenSearch cluster with more than one node. Let's use the sample [docker-compose.yml](https://opensearch.org/samples/docker-compose.yml) to start a cluster with two nodes. The cluster's API will be available at `localhost:9200` with basic authentication enabled with default username and password of `admin:admin`.
+In this guide, we will need an OpenSearch cluster with more than one node. Let's use the sample [docker-compose.yml](https://opensearch.org/samples/docker-compose.yml) to start a cluster with two nodes. The cluster's API will be available at `localhost:9200` with basic authentication enabled with default username and password of `admin:< Admin password >`.
```
Comment (dblock) [inline guides/index_lifecycle.md:7]: Lowercase Admin since the username is admin?

Comment (derek-ho) [reply guides/index_lifecycle.md]: Done in [d6f5ec4](https://github.com/opensearch-project/opensearch-go/pull/449/commits/d6f5ec4a64cd1ac36bfa017eacde24de127673f8)

**Comment by derek-ho** [comment]
@dblock can you take another look here? I am not sure why the tests against the latest secure cluster are passing, since the password changes should not be there? I also think the unit test failure is unrelated?

File: .ci/opensearch/Dockerfile.opensearch [hunk 2]
```diff
@@ -5,9 +5,15 @@ ARG opensearch_path=/usr/share/opensearch
 ARG SECURE_INTEGRATION
 ENV SECURE_INTEGRATION=$SECURE_INTEGRATION
 
+# Starting in 2.12.0 security demo requires an initial admin password, which is set as myStrongPassword123!
+ENV password="admin"
+RUN if [ "$OPENSEARCH_VERSION" = "latest" ]; then \
+        export password="myStrongPassword123!"; \
+    fi
```
Comment (dblock) [inline .ci/opensearch/Dockerfile.opensearch:12]: This will break immediately after 2.12 is released.

**Review by dblock — CHANGES_REQUESTED** [review]
Is this change necessary at all until we udate CI/CD to 2.12?

File: .ci/opensearch/Dockerfile.opensearch (hunk 2 above)
Comment (DarshitChanpura) [reply .ci/opensearch/Dockerfile.opensearch:12]: this should be updated to check for versions 2.12 and above

**Comment by DarshitChanpura** [comment]
@dblock @VijayanB @VachaShah @Jakob3xD  Now that 2.12 is released this should be unblocked. Would one of you mind bringing this home?

**Comment by VachaShah** [comment]
> @dblock @VijayanB @VachaShah @Jakob3xD Now that 2.12 is released this should be unblocked. Would one of you mind bringing this home?

Sure! @derek-ho Is this ready for review now?

File: .ci/opensearch/Dockerfile.opensearch [hunk 3]
```diff
@@ -5,9 +5,15 @@ ARG opensearch_path=/usr/share/opensearch
 ARG SECURE_INTEGRATION
 ENV SECURE_INTEGRATION=$SECURE_INTEGRATION
 
+# Starting in 2.12.0 security demo requires an initial admin password, which is set as myStrongPassword123!
+ENV password="admin"
+RUN if [ "$OPENSEARCH_VERSION" = "latest" ]; then \
```
Comment (derek-ho) [inline .ci/opensearch/Dockerfile.opensearch:10]: Add version logic here

File: opensearch_secure_integration_test.go [hunk 4]
```diff
@@ -37,11 +40,22 @@ import (
 )
 
 func getSecuredClient() (*opensearchapi.Client, error) {
+	version := os.Getenv("OPENSEARCH_VERSION")
+	parts := strings.Split(version, ".")
```
Comment (Jakob3xD) [inline opensearch_secure_integration_test.go:44]: We have a [ParseVersion function](https://github.com/opensearch-project/opensearch-go/blob/main/opensearch.go#L239) that can be used.

**Comment by derek-ho** [comment]
@Jakob3xD thanks. I'm getting a little stuck on this PR since it seems by the time we reach here we have no context in regards to the OpenSearch version, since that is spun up as a different step. Any ideas?

**Comment by Jakob3xD** [comment]
> @Jakob3xD thanks. I'm getting a little stuck on this PR since it seems by the time we reach here we have no context in regards to the OpenSearch version, since that is spun up as a different step. Any ideas?

I hope it is okay that I force pushed into your branch. I adjusted the docker file to fix the health check and change the go security check to find the password by trying them out. Otherwise we need to adjust the workflow to export the opensearch version as env so we can parse it as you tried before.

**Review by dblock — APPROVED** [review]

**Pull request merged**